lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -3025,6 +3025,83 @@ static void printVerifAttrs(OpAsmPrinter &p, Operation *op,
   printElideEmptyName(p, op, attr, {"message"});
 }
 
+//===----------------------------------------------------------------------===//
+// NonLocalAnchor helpers.
+//===----------------------------------------------------------------------===//
+
+bool NonLocalAnchor::dropModule(StringAttr moduleToDrop) {
+  SmallVector<Attribute, 4> newPath;
+  bool updateMade = false;
+  for (auto nameRef : namepath()) {
+    // nameRef is either an InnerRefAttr or a FlatSymbolRefAttr.
+    if (auto ref = nameRef.dyn_cast<hw::InnerRefAttr>()) {
+      if (ref.getModule() == moduleToDrop) {
+        updateMade = true;
+        continue;
+      } else
+        newPath.push_back(ref);
+    } else {
+      if (nameRef.cast<FlatSymbolRefAttr>().getAttr() == moduleToDrop) {
+        updateMade = true;
+        break;
+      } else
+        newPath.push_back(nameRef);
+    }
+  }
+  if (updateMade)
+    namepathAttr(ArrayAttr::get(getContext(), newPath));
+  return updateMade;
+}
```

Comment:
A flavor of dropping is inlining. For this, the names are concatenated with an interleaved `_`. Perhaps this method should take an option to enable this behavior.

Reply:
Added another inline flavor. assuming that the symbol name of the corresponding instance inside the inlined module gets prepended with the symbol name of the instance where it got inlined.
so, if `test1` gets inlined at, 
`firrtl.instance a1 sym @sym1 @test1`
```
firrtl.module @test1() {
firrtl.instance b1 sym @sym3 @test3
}
 ```
then we assume, result is,
```
firrtl.instance a1_b1 sym @sym1_sym3 @test3

 ```

Reply:
Nice. In retrospect, I had a wrong assumption that the symbol was going to get updated. Previously, the _name_ was going to get inlined and this utility made sense. In the new `hw::InnerRefAttr` world, there is no reason for the symbol to change to match an inlined name (except in the case where inlining results in a module namespace name collision).